### PR TITLE
[VEN-2016][VEN-2017][VEN-2050][VEN-2052][VEN-2053][VEN-2055]: Code4rena findings fixes for Prime

### PR DESCRIPTION
**Pull request by chechu**
## Description

Resolves #VEN-2016 and VEN-2017
VEN-2052: 13a5c7e5a721a85d19eeb609d28c30e2ecf03f9b

## Checklist

- [ ] I have updated the documentation to account for the changes in the code.
- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
- [ ] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Debugger022** [comment]
Commit hash for VEN-2052 is 13a5c7e5a721a85d19eeb609d28c30e2ecf03f9b

File: contracts/Tokens/Prime/Prime.sol [hunk 1]
```diff
@@ -12,13 +12,18 @@ import { PrimeStorageV1 } from "./PrimeStorage.sol";
...
 import { IVToken } from "./Interfaces/IVToken.sol";
 import { InterfaceComptroller } from "./Interfaces/InterfaceComptroller.sol";
 
-
-/// @custom:security-contact https://github.com/VenusProtocol/venus-protocol
-contract Prime is AccessControlledV8, PausableUpgradeable, MaxLoopsLimitHelper, PrimeStorageV1 {
+/**
+ * @title Prime
+ * @author Venus
+ * @notice Prime Token is used to provide extra rewards to the users who have staked a minimum of 1000 XVS in the XVSVault for 90 days
```
Comment (chechu) [inline contracts/Tokens/Prime/Prime.sol:23]:
```suggestion
 * @notice Prime Token is used to provide extra rewards to the users who have staked a minimum of `MINIMUM_STAKED_XVS` XVS in the XVSVault for `STAKING_PERIOD` days
```

File: contracts/Tokens/Prime/Prime.sol [hunk 2]
```diff
@@ -251,11 +259,14 @@ contract Prime is AccessControlledV8, PausableUpgradeable, MaxLoopsLimitHelper,
...
      * @custom:event Emits UserScoreUpdated event
      */
-    function updateScores(address[] memory users) external {
+    function updateScores(address[] calldata users) external {
         if (pendingScoreUpdates == 0) revert NoScoreUpdatesRequired();
         if (nextScoreUpdateRoundId == 0) revert NoScoreUpdatesRequired();
 
-        for (uint256 i = 0; i < users.length; ) {
+        uint256 usersLength = users.length;
+        _ensureMaxLoops(usersLength);
```
Comment (chechu) [inline contracts/Tokens/Prime/Prime.sol:267]: What is the benefit of this check? Saving gas if the transaction is going to be reverted due to out-of-gas? I think I would add this check and I would leave the transaction to fail (I think it should/will be detected by the client/simulation invoking the function)

File: contracts/Tokens/Prime/Prime.sol [hunk 3]
```diff
@@ -404,8 +434,11 @@ contract Prime is AccessControlledV8, PausableUpgradeable, MaxLoopsLimitHelper,
     function issue(bool isIrrevocable, address[] calldata users) external {
         _checkAccessAllowed("issue(bool,address[])");
 
+        uint256 usersLength = users.length;
+        _ensureMaxLoops(usersLength);
```
Comment (chechu) [inline contracts/Tokens/Prime/Prime.sol:438]: I think I wouldn't add this check here, and I would leave the transaction to fail due to out-of-gas

File: contracts/Tokens/Prime/Prime.sol [hunk 4]
```diff
@@ -779,32 +859,35 @@ contract Prime is AccessControlledV8, PausableUpgradeable, MaxLoopsLimitHelper,
...
             unchecked {
                 ++i;
             }
         }
 
-        if (tokens[user].isIrrevocable) {
+        if (token.isIrrevocable) {
             totalIrrevocable--;
         } else {
             totalRevocable--;
```
Comment (chechu) [inline contracts/Tokens/Prime/Prime.sol:886]: pre-decrement?

File: contracts/Tokens/Prime/Prime.sol [hunk 5]
```diff
@@ -268,8 +268,8 @@ contract Prime is IPrime, AccessControlledV8, PausableUpgradeable, MaxLoopsLimit
     function updateScores(address[] calldata users) external {
         if (pendingScoreUpdates == 0) revert NoScoreUpdatesRequired();
         if (nextScoreUpdateRoundId == 0) revert NoScoreUpdatesRequired();
-
-        for (uint256 i; i < usersLength; ) {
+
```
Comment (chechu) [inline contracts/Tokens/Prime/Prime.sol:271]: Shouldn't we keep `uint256 usersLength = users.length;`

File: contracts/Tokens/Prime/Prime.sol [hunk 6]
```diff
@@ -462,7 +462,7 @@ contract Prime is IPrime, AccessControlledV8, PausableUpgradeable, MaxLoopsLimit
         _checkAccessAllowed("issue(bool,address[])");
```
Comment (chechu) [inline contracts/Tokens/Prime/Prime.sol:463]: Shouldn't we keep `uint256 usersLength = users.length;`?

File: contracts/Tokens/Prime/Prime.sol (hunk 1 above)
Comment (chechu) [reply contracts/Tokens/Prime/Prime.sol:23]: Changes have been undo

**Review by chechu — COMMENTED** [review]
LGTM

**Pull request merged**